mods/default/nodes.lua

```diff
@@ -1986,6 +1986,45 @@ minetest.register_node("default:sand_with_kelp", {
 -- Corals
 --
 
+local function coral_on_place(itemstack, placer, pointed_thing)
+	if pointed_thing.type ~= "node" or not placer then
+		return itemstack
+	end
+
+	local player_name = placer:get_player_name()
+	local pos_under = pointed_thing.under
+	local pos_above = pointed_thing.above
+	local node_under = minetest.get_node(pos_under)
+	local def_under = minetest.registered_nodes[node_under.name]
+
+	if def_under and def_under.on_rightclick then
+		return def_under.on_rightclick(pos_under, node_under.name,
+				placer, itemstack, pointed_thing) or itemstack
```

Comment:
The condition ensures that `def_under.on_rightclick` is not nil, so why use `or itemstack`? That is never needed?

Reply:
[`core.item_place` does this, too.](https://github.com/minetest/minetest/blob/5560ec8178053b27adac6f277f539975d46267b8/builtin/game/item.lua#L424) Even though it is not implicitly allowed, there might be some mods which simply don't return the itemstack in `on_rightclick`. And normally there won't be any issues with that.

Reply:
Ah i see, my mistake. Is fine then.